syntax = "proto2";

message Discover {
    optional string device_type = 1;
    optional string communication_type = 2;
    optional string ip = 3;
    optional int32 port = 4;
}

message Data {
    optional string data = 1;
}

message Command {
    optional string command = 1;
    repeated string arguments = 2;
}

message CommandResponse {
    optional bool status = 1;
    optional string message = 2;
}

message Device {
    optional int32 id = 1;
    optional string device_type = 2;
    optional string communication_type = 3;
}

message DeviceList {
    repeated Device devices = 1;
}

message Message {
    optional string type = 1;
    oneof content {
        Discover discover = 2;
        Data data = 3;
        Command command = 4;
        CommandResponse command_response = 5;
        Device device = 6;
        DeviceList device_list = 7;
    }
}